File: module/module_mvp/src/main/java/com/xiaochen/module/mvp/api/IServiceApi.java
package com.xiaochen.module.mvp.api;
import com.xiaochen.module.mvp.response.HomeArticleRespVO;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * <p>服务器接口{d}</p>
 *
 * @author zhenglecheng
 * @date 2019-12-08
 */
public interface IServiceApi {

    /**
     * 首页文章列表
     */
    @GET("article/list/{page}/json")
    Observable<HomeArticleRespVO> getHomeArticles(@Path("page") int page);
}

File: module/module_mvvm/src/main/java/com/xiaochen/module/mvvm/api/IServiceApi.kt
package com.xiaochen.module.mvvm.api

import com.xiaochen.module.mvvm.response.HomeArticleRespVO
import retrofit2.http.GET
import retrofit2.http.Path

/**
 *
 * 服务器接口{d}
 *
 * @author zhenglecheng
 * @date 2019-12-08
 */
interface IServiceApi {

    /**
     * 首页文章列表
     */
    @GET("/article/list/{page}/json")
    suspend fun getHomeArticles(@Path("page") page: Int): HomeArticleRespVO
}
